File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        n = len(temperatures)
        next_warmer_in = [0] * n

        stack = []

        for i in range(n-1, -1, -1):
            while stack and temperatures[stack[-1]] <= temperatures[i]:
                stack.pop()


            if not stack:
                next_warmer_in[i] = 0
            else:
                next_warmer_in[i] = abs(stack[-1] - i)
            
            stack.append(i)
        

        return next_warmer_in

File: 0739-daily-temperatures/0739-daily-temperatures.go
func dailyTemperatures(temperatures []int) []int {
    n := len(temperatures)
    res := make([]int , n)

    stack := make([]int , 0, n)

    for i, v := range temperatures {
        for len(stack) > 0 && v > temperatures[stack[len(stack)-1]] {
            top := stack[len(stack) - 1]

            res[top] = i - top

            stack = stack[:len(stack)-1]
        }

        stack = append(stack, i)
    }

    return res
}